packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -68,14 +73,140 @@ import { StorageServiceWithAttachBlobs } from "./storageServiceWithAttachBlobs";
 import { IDataStoreAliasMessage, channelToDataStore, isDataStoreAliasMessage } from "./dataStore";
 import { GCNodeType, detectOutboundRoutesViaDDSKey } from "./gc";
 import { IContainerRuntimeMetadata, nonDataStorePaths, rootHasIsolatedChannels } from "./summary";
+import { ContainerMessageType, LocalContainerRuntimeMessage } from "./messageTypes"
+import { FluidDataStoreRegistry } from "./dataStoreRegistry";
+import { trimLeadingAndTrailingSlashes } from "./gc";
+
+/**
+ * Accepted header keys for requests coming to the runtime.
+ * @internal
+ */
+export enum RuntimeHeaders {
+	/** True to wait for a data store to be created and loaded before returning it. */
+	wait = "wait",
+	/** True if the request is coming from an IFluidHandle. */
+	viaHandle = "viaHandle",
+}
+
+/** True if a tombstoned object should be returned without erroring
+ * @alpha
+ */
+export const AllowTombstoneRequestHeaderKey = "allowTombstone"; // Belongs in the enum above, but avoiding the breaking change
+/**
+ * [IRRELEVANT IF throwOnInactiveLoad OPTION NOT SET] True if an inactive object should be returned without erroring
+ * @internal
+ */
+export const AllowInactiveRequestHeaderKey = "allowInactive"; // Belongs in the enum above, but avoiding the breaking change
 
 type PendingAliasResolve = (success: boolean) => void;
 
+interface FluidDataStoreMessage {
+	content: any;
+	type: string;
+}
+
+export function cloneParentContext(parentContext: IFluidParentContext): IFluidParentContext {
```

Comment:
#todo 

- rename this
- explore a better way of having it get a set of properties to override the defaults

Reply:
Exploring JS proxy objects might be interesting here. That said I do not understand fully implications and management of identity in proxies.
The only thing I can say that {...object} does not look right. I've started with Object.assign({}, parent), but something was not working about it (and in general, I have a hard time thinking about what I get as result of it - i.e. I do not understand implications of cloning full ContainerRuntime class that way).
So this pattern felt really safe to me, even though verbose. You might find better ways of doing it!